airbyte-ci/connectors/connector_ops/connector_ops/utils.py

```diff
@@ -325,6 +325,35 @@ def get_local_dependency_paths(self, with_test_dependencies: bool = True) -> Set
                 self.code_directory / "build.gradle", with_test_dependencies=with_test_dependencies
             )
         return sorted(list(set(dependencies_paths)))
+    
+
+@dataclass(frozen=True)
+class CDK:
+    """Utility class to gather metadata about a cdk."""
+
+    language: str
+
+    @property
+    def code_directory(self) -> Path:
+        return Path(f"./airbyte-cdk/{self.language}")
+
+
+    @property
+    def version(self) -> str:
```

Comment:
I think it's better to set it as instantiation time instead of using a property.
If you want to keep using a dataclass I recommend using the `__post_init__` hook: https://docs.python.org/3/library/dataclasses.html#post-init-processing

Reply:
👍